packages/nft-erc721-provider/lib/NftErc721Provider.ts

```diff
@@ -0,0 +1,62 @@
+import { Provider } from '@liquality/provider'
+import { addressToString } from '@liquality/utils'
+import { ensure0x, normalizeTransactionObject } from '@liquality/ethereum-utils'
+import { NftProvider, Address, BigNumber } from '@liquality/types'
+
+import { Contract } from '@ethersproject/contracts'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { Wallet } from '@ethersproject/wallet'
+
+import NftErc721_ABI from './NftErc721_ABI.json'
+
+export default class NftErc721Provider extends Provider implements Partial<NftProvider> {
+  _contract: Contract
+  _wallet: Wallet
+  _jsonRpcProvider: JsonRpcProvider
+
+  constructor(options: { uri: string; mnemonic: string; derivationPath: string }) {
+    super()
+
+    this._wallet = Wallet.fromMnemonic(options.mnemonic, options.derivationPath)
+    this._wallet = this._wallet.connect(new JsonRpcProvider(options.uri))
```

Comment:
An ethers Provider will execute frequent getNetwork calls to ensure the network calls and network being communicated with are consistent.
In our case this is not the case, so you can use `StaticJsonRpcProvider` instead.